usr/local/www/diag_traceroute.php

```diff
@@ -46,15 +46,35 @@
 $allowautocomplete = true;
 $pgtitle = array(gettext("Diagnostics"),gettext("Traceroute"));
 include("head.inc");
-
-?>
-<body link="#0000CC" vlink="#0000CC" alink="#0000CC">
-<?php include("fbegin.inc"); ?>
-<?php
-
+	
 define('MAX_TTL', 64);
 define('DEFAULT_TTL', 18);
 
+$pconfig['ttl'] = DEFAULT_TTL;
+$pconfig['ipproto'] = 'IPv4';
+$pconfig['sourceip'] = 'Any';
+
+function create_sourceaddresslist() {
+	$list = array('any' => 'Any');
+
+	$sourceips = get_possible_traffic_source_addresses(true);
+
+	foreach ($sourceips as $sipvalue => $sipname):
```

Comment:
I wouldn't use ```foreach: endforeach;``` here; since it's a single line you can ditch the ```: endforeach;``` or add accolades